rpcserver.go

```diff
@@ -7687,13 +7687,29 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 	}
 	iReq := req.InvoiceRequest
 
-	// Do some preliminary checks on the asset ID and make sure we have any
-	// balance for that asset.
-	if len(req.AssetId) != sha256.Size {
-		return nil, fmt.Errorf("asset ID must be 32 bytes")
+	var specifier asset.Specifier
+
+	switch {
+	case len(req.AssetId) > 0 && len(req.GroupKey) > 0:
```

Comment:
Same here and below re: using existing functions and/or extracting into function.